taui/src/components/dock.js

```diff
@@ -52,25 +69,196 @@ export default class Dock extends PureComponent<Props> {
+      } else {
+        remove(favorites, f => f === neighborhoodId)
+      }
+    }
+
+    profile.favorites = favorites
+    const isAnonymous = !profile || profile.key === ANONYMOUS_USERNAME
+
+    if (!isAnonymous) {
+      Storage.put(profile.key, JSON.stringify(profile))
+        .then(result => {
+          changeUserProfile(profile)
+        })
+        .catch(err => {
+          console.error(err)
+        })
+    } else {
+      // Do not attempt to write anonymous profile to S3
+      changeUserProfile(profile)
+    }
+  }
+
+  // Toggle between showing all and showing favorites, if not already in the new state
+  showAll (show: boolean) {
+    if (show !== this.state.showAll) {
+      this.setState({page: 0, showAll: show})
+    }
+  }
+
+  // Render list pagination buttons, or button to return to list from details
+  buttonRow (props) {
+    const {
+      haveAnotherPage,
+      page,
+      showDetails
+    } = props
+
+    const backFromDetails = this.backFromDetails
+    const goPreviousPage = this.goPreviousPage
+    const goNextPage = this.goNextPage
+
+    return (
+      <div className='account-profile__destination_row'>
+        {showDetails && <button
+          className='account-profile__button account-profile__button--secondary account-profile__destination_narrow_field'
+          onClick={backFromDetails}>{message('Dock.GoBackFromDetails')}
+        </button>}
+        {!showDetails && page > 0 && <button
+          className='account-profile__button account-profile__button--secondary account-profile__destination_narrow_field'
+          onClick={goPreviousPage}>{message('Dock.GoPreviousPage')}
+        </button>}
+        {!showDetails && haveAnotherPage && <button
+          className='account-profile__button account-profile__button--secondary account-profile__destination_narrow_field'
+          onClick={goNextPage}>{message('Dock.GoNextPage')}
+        </button>}
+      </div>)
+  }
+
+  // Render list of neighborhoods
+  neighborhoodsList (props) {
     const {
       activeNetworkIndex,
+      changeUserProfile,
+      neighborhoods,
+      setActiveNeighborhood,
+      setFavorite,
+      startingOffset,
+      userProfile
+    } = props
+    const goToDetails = this.goToDetails
+    const neighborhoodPage = neighborhoods && neighborhoods.length ? neighborhoods.slice(
+      startingOffset, SIDEBAR_PAGE_SIZE + startingOffset) : []
+
+    if (!neighborhoodPage || !neighborhoodPage.length) {
+      return <div className='Card'>{message('Dock.NoResults')}</div>
+    }
+
+    return (
+      neighborhoodPage.map((neighborhood, index) =>
+        neighborhood.segments && neighborhood.segments.length
+          ? (<RouteCard
+            cardColor={neighborhood.active ? 'green' : NETWORK_COLORS[activeNetworkIndex]}
+            goToDetails={goToDetails}
+            index={index}
+            isFavorite={userProfile.favorites.indexOf(neighborhood.properties.id) !== -1}
```

Comment:
After creating a new profile, I kept getting a blank map with a console error:
```
TypeError: userProfile.favorites is undefined[Learn More] dock.js:178:12
    neighborhoodsList dock.js:178
    map (index):286
    neighborhoodsList dock.js:172
    neighborhoodsList (index):976
    React 6
    wrapped build.umd.js:2357
    React 12
    onStateChange (index):972
    Redux 3
    promiseMiddleware promise.js:18
    multiMiddleware multi.js:12
    middleware fetch.js:179
    routerMiddleware React
    dispatch Redux
    map (index):286
    multiMiddleware multi.js:12
    middleware fetch.js:179
    routerMiddleware React
    dispatch Redux
    promiseMiddleware promise.js:17
```
until I added `userProfile && ` to this line. After I did that and added a few favorites, everything worked fine, even if I reverted the change and deleted all favorites. So I think the new profile creation code might not be properly initializing the favorites.

Reply:
Updated setting the default.